docs/development/text-handling.md

```diff
@@ -0,0 +1,278 @@
+These are plain text (as this is a baseline that is supported in every interesting
+target context) and HTML (since this is the markup for browsers, the main environment
+of our users). These conversion can be lossful, though.
+* It may be possible to convert some text in a given representation to some other
+representation but in general it is only expected that every text can be converted
+to HTML and plain text.
+
+
+## Approach
+
+To solve the requirements we are looking to implement the following approach in
+a sub folder of `src\Data`, using the conventions and standards of that library.
+Conversions will also be made available via the `Refinery` to integrate them
+into an established framework.
+
+### Define Structure Options
+
+To make it possible to programmatically talk about structuring options for text,
+a central `class` (and later `enum`, once supported for PHP < 8.1 is cut) defines
+the options that we care about:
+
+```php
+class Structure
+{
+    const string BOLD = "bold";
+    const string ITALIC = "italic";
+    const string UNORDERED_LIST = "unordered_list";
+    /* ... */
+}
+```
+
+### Define Markup
+
+Text will be represented as `string` in memory. Since we do not care about the
+information or specific structure of a given text, it seems to be unexpected that
+texts need to be represented as abstract syntax trees or something alike. Text
+might temporarily be transformed to non-`string` representations during conversions
+from one representation to another, but these representations will be local to the
+conversion.
+
+The set of available markup will likely be mostly static. The different markup
+classes might become carrier for markup specific methods (e.g. escaping...). At
+the current state of this proposal it is not clear if a shared `interface` for
+`Markup` classes can or should have common methods or are just a tag.
+
+```php
+interface Markup
+{
+}
+
+class HTML implements Markup
+{
+    public function restrictUsedTags(string $in, array $tags) : string; // for example
+}
+```
+
+### Define Shapes for Text
+
+These will be the workhorses for the toolset we propose. A shape bundles
+information about markup and structuring options. It can produce text data from
+raw string input and convert given data to other shapes. We expect that there
+will be families of shapes that share most of their code via class hierarchies.
+A markdown family, e.g., could contain various markdown shapes with the same
+representation but different structuring options.
+
+```php
+interface Shape 
+{
+    /**
+     * @throws \InvalidArgumentException if $text does not match format. 
+     */
+    public function toHTML($text) : HTMLText;
+
+    /**
+     * @throws \InvalidArgumentException if $text does not match format. 
+     */
+    public function toPlainText($text) : PlainText;
+
+    public function getMarkup() : Markup;
+
+    /**
+     * @return TextStructure[]
```

Comment:
Could give an example wthat this type could be and which kind of data it encapsulates?

Reply:
This is just wrong. Currently this would be consts from `Structure`. It could become an `enum` at some point. Will correct...

Reply:
:heavy_check_mark: